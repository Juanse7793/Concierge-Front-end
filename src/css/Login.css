/* LoginPage */
.login {
  background: #fff;
  display: flex;
  align-items: flex-start;
  border: 4px solid #fff;
  border-radius: 20px;
  padding: 2.5rem;
  max-width: 95vw;
}

.login img {
  margin: 0 1rem 0 0;
}

.login h1 {
  margin-top: 0;
}

#sign-btn {
  margin: 0.5rem 0 1rem 0;
}

#sign-btn:hover {
  background: #fff;
  color: var(--theme-lime);
}

.error {
  color: var(--theme-red);
  text-align: right;
  margin: -24px 4px 0 0;
}

@media (max-width: 768px) {
  .login {
    flex-direction: column;
    align-items: center;
  }

  .login img {
    margin: -1rem 0 0 0;
  }
}
