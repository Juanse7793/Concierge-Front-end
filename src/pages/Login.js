import React from 'react';
import { useDispatch } from 'react-redux';
import Button from '@mui/material/Button';
import CssBaseline from '@mui/material/CssBaseline';
import TextField from '@mui/material/TextField';
import FormControlLabel from '@mui/material/FormControlLabel';
import Checkbox from '@mui/material/Checkbox';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Container from '@mui/material/Container';
import { NavLink } from 'react-router-dom';
import { signIn } from '../redux/reducers/users';

export default function Login() {
  const dispatch = useDispatch();
  const handleSubmit = (event) => {
    event.preventDefault();
    const data = new FormData(event.currentTarget);
    const username = data.get('username');
    dispatch(signIn(username));
  };

  return (
    <div className="log">
      <Container component="main" maxWidth="xs" className="comp">
        <CssBaseline />
        <Box
          sx={{
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
          }}
        >
          <img src="concierge.png" alt="logo" />
          <Typography component="h1" variant="h5">
            Sign In
          </Typography>
          <Box component="form" onSubmit={handleSubmit} noValidate sx={{ mt: 1 }}>
            <TextField
              margin="normal"
              required
              fullWidth
              id="username"
              label="Username"
              name="username"
              autoComplete="username"
              autoFocus
            />
            <FormControlLabel
              control={<Checkbox value="remember" color="primary" />}
              label="Remember me"
            />
            <Button
              type="submit"
              fullWidth
              id="sign-btn"
              variant="contained"
              sx={{ mt: 3, mb: 2 }}
            >
              Sign In
            </Button>
            <Grid container>
              <Grid item>
                <NavLink activeClassName="is-active" to="/sign-up">
                  Dont have an account? Sign Up
                </NavLink>
              </Grid>
            </Grid>
          </Box>
        </Box>
      </Container>
    </div>

  );
}
