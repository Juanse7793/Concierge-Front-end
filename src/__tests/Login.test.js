import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Login from '../pages/Login';

const mockStore = configureMockStore();
const store = mockStore({ user: { error: false } });

it('matches snapshot', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Router>
        <Login />
      </Router>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
