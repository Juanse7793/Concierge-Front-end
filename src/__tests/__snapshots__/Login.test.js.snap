// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="background row"
>
  <div
    className="login row"
  >
    <img
      alt="logo"
      src="concierge.png"
    />
    <form
      className="column"
      onSubmit={[Function]}
    >
      <h1>
        Sign
         In
      </h1>
      <div
        className="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor=":r0:"
          id=":r0:-label"
        >
          Username
          <span
            aria-hidden={true}
            className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="username"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id=":r0:"
            name="username"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1ftyaf0"
            >
              <span>
                Username
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            onChange={[Function]}
            type="checkbox"
            value="remember"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Remember me
        </span>
      </label>
      <button
        className="pill green"
        id="sign-btn"
        type="submit"
      >
        Sign
         
         In
      </button>
      <a
        href="/sign-up"
        onClick={[Function]}
      >
        Don't have an account? Sign Up
      </a>
    </form>
  </div>
</section>
`;
